inputters/xml.lua

```diff
@@ -74,15 +74,7 @@ function inputter.parse (_, doc)
   if not tree then
     SU.error(err)
   end
-  -- XML documents can have any root element, and it should be up to the class
-  -- to supply handling far whatever that element that is in a specific format.
-  -- Hence we wrap the actual DOM in an extra element of our own if and only if
-  -- it doesn't look like a native SILE one already.
-  local root = tree.command
-  if root ~= "sile" and root ~= "document" then
-    tree = { tree, command = "document" }
-  end
-  return { tree }
+  return tree
```

Comment:
I can understand why there was a "document" level wrapped around the "tree".
Say you have an XML file not containing a `<document>` or `<sile>` root, but its own (schema-specific) root tag.
How to process it with SILE?
I'd say, possibly[^1], with something such as:
```
sile -c myxmlclass myxmlfile.xml
```
Before, this was applied to the `<document>` pseudo-encapsulation.
If it is removed, it will apply to the top-level tag, but possibly doing weird things with its own attributes.

[^1]: I formerly used the `-I` preamble, and am still at loss finding an appropriate replacement.

Reply:
In what scenario would an input document have something as a top level tag that we should reject entirely and not process as the top level AST element that brings with it the class instantialization?

Reply:
I don't get your question.
The previous code was not rejecting anything, it was adding a document level when the root tag isn't "sile" or "document". In other terms, given an XML document:
```
<tei>
  <!-- TEI dictionary stuff -->
 </tei>
 ```
Then SILE regarded it as wrapped in a `<document>`:
```
$ sile mydict.xml
SILE v0.14.4 (Lua 5.2)
<xml.xml> as xml

! Unknown command tei at xml.xml:1:1
```
And assuming "myclass" defines all the necessary tags, `sile -c myclass mydict.xml` worked.

The `<document>` encapsulation was required, for `inputter:requireClass()` to work as expected.

Reply:
You're right, I lost track of the fact that classes don't provide the XML input parser in those cases and they may *want* to wrap the whole document.